Scripts and tools: fix gitian build related to Debian/MacOSX-SDK/curl

cisba opened the pull request:
The [documentation](https://github.com/bitcoin-core/docs/blob/master/gitian-building.md) suggest to use debian 8.x, but the script bin/make-base-vm is invoked with '--suite bionic’ that on debian rises an error (not easy to understand for a newby). It is solved detecting automatically the distro and the suite-codename invoking lsb_release.

The [documentation](https://github.com/bitcoin-core/docs/blob/master/gitian-building/gitian-building-mac-os-sdk.md) instruct to create the directory ‘gitian-builder/inputs’ with the MacOSX  SDK inside. The presence of this path does not consent to clone the gitian-builder repos and rises a subsequent error not clearly related to the issue. It is solved checking the presence of the ‘.git’ inside and moving the SDK temporary outside, relocating at his place after cloning.

The curl package is required for the build process as described [here](https://github.com/bitcoin/bitcoin/blob/master/depends/README.md).

cisba committed:
Scripts and tools: fix two gitian build issues related to Debian and MacOSX SDK

The [documentation](https://github.com/bitcoin-core/docs/blob/master/gitian-building.md) suggest to use debian 8.x, but the script bin/make-base-vm is invoked with '--suite bionic’ that on debian rises an error (not easy to understand for a newby). It is solved detecting automatically the distro and the suite-codename invoking lsb_release.

The [documentation](https://github.com/bitcoin-core/docs/blob/master/gitian-building/gitian-building-mac-os-sdk.md) instrunct to create the directory ‘gitian-builder/inputs’ with the MacOSX  SDK inside. The presence of this path does not consent to clone the gitian-builder repos and rises a subsequent error not clearly related to the issue. It is solved checking the presence of the ‘.git’ inside and moving the SDK temporary outside, relocating at his place after cloning.

hebasto commented on contrib/gitian-build.py:
```diff
@@ -26,12 +27,18 @@ def setup():
-    make_image_prog = ['bin/make-base-vm', '--suite', 'bionic', '--arch', 'amd64']
+    distro = subprocess.check_output(['lsb_release', '-is']).decode("utf-8").strip('\n').lower()
+    suite = subprocess.check_output(['lsb_release', '-cs']).decode("utf-8").strip('\n')
+    make_image_prog = ['bin/make-base-vm', '--distro', distro, '--suite', suite, '--arch', 'amd64']
```
Do not move from `bionic`. Ref: #12511

cisba replied:
Ah... ok, then I will PR to correct the documentation to use ubuntu/bionic.
Thanks @hebasto !
And debian/stretch?

ken2812221 replied:
> The documentation suggest to use debian 8.x, but the script bin/make-base-vm is invoked with '--suite bionic’ that on debian rises an error 

You should upgrade your lxc >= 2.1.1 and debootstrap >= 1.0.92

achow101 replied:
Do not change this line, revert your changes. The VM that is built is for the gitian build and it's distro and suite have nothing to do with the host OS. The same distro and suite must be used in order to have the same binaries be built.

cisba replied:
gitianuser@debian:~$ dpkg-query -W lxc debootstrap
debootstrap	1.0.89
lxc	1:2.0.7-2+deb9u2
gitianuser@debian:~$ lsb_release -a
No LSB modules are available.
Distributor ID:	Debian
Description:	Debian GNU/Linux 9.5 (stretch)
Release:	9.5
Codename:	stretch

@ken2812221  It seem that debian 9.x does not provide it...

MarcoFalke replied:
@cisba You can use ubuntu bionic or self-compile and install them from source.

MarcoFalke replied:
See the note in section https://github.com/bitcoin-core/docs/blob/master/gitian-building.md#preparing-the-gitian-builder-host

cisba replied:
@MarcoFalke I followed it but does not work because bionic is hard coded in gitian scripts.
I tried changing it but others errors rise so I quit for a wile debian vm.

cisba replied:
@hebasto comment is the more remarkable IMHO

cisba committed:
Scripts and tools: fix missed curl in setup

The curl package is required for the build process as described [here](https://github.com/bitcoin/bitcoin/blob/master/depends/README.md).

achow101 commented:
NACK. The changes constitute a misunderstanding of how Gitian works and are not an improvement.

> The documentation suggest to use debian 8.x, but the script bin/make-base-vm is invoked with '--suite bionic’ that on debian rises an error (not easy to understand for a newby). It is solved detecting automatically the distro and the suite-codename invoking lsb_release.

As mentioned in the comments, this is not how gitian works. The Gitian VM must use specifically Ubuntu Bionic. It is completely independent of the host OS.

> The documentation instruct to create the directory ‘gitian-builder/inputs’ with the MacOSX SDK inside. The presence of this path does not consent to clone the gitian-builder repos and rises a subsequent error not clearly related to the issue. It is solved checking the presence of the ‘.git’ inside and moving the SDK temporary outside, relocating at his place after cloning.

The Gitian environment should be setup before fetching the SDK. The documentation should be fixed to reflect that.

> The curl package is required for the build process as described here.

That is done inside of the VM itself. Curl is not needed for setting up and starting gitian.

cisba commented:
Thanks @achow101 for pointing me in the right direction.
All the issues are caused by documentation. 
I'll try to fix docs.
